compiler/pom.xml

```diff
@@ -173,6 +156,41 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>2.3</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <minimizeJar>true</minimizeJar>
+              <artifactSet>
+                <excludes>
+                  <!-- guava which has a consistent API and whose public types we vend in producers -->
+                  <exclude>com.google.guava</exclude>
```

Comment:
Yep. We've had this happen multiple times when used on the Gradle
buildscript classpath.

On Thu, Apr 2, 2015, 7:22 AM Roman Mazur <notifications@github.com> wrote:

> In compiler/pom.xml
> <https://github.com/google/dagger/pull/118#discussion_r27648909>:
>
> > +      <plugin>
> > +        <groupId>org.apache.maven.plugins</groupId>
> > +        <artifactId>maven-shade-
>
> plugin</artifactId>
> > +        <version>2.3</version>
> > +        <executions>
> > +          <execution>
> > +            <phase>package</phase>
> > +            <goals>
> > +              <goal>shade</goal>
> > +            </goals>
> > +            <configuration>
> > +              <minimizeJar>true</minimizeJar>
> > +              <artifactSet>
> > +                <excludes>
> > +                  <!-- guava which has a consistent API and whose public types we vend in producers -->
> > +                  <exclude>com.google.guava</exclude>
>
> Actually this might cause problems. For example, Android gradle plugin has
> a transient dependency on guava 15. And when some plugin has a dependency
> on guava 18, it apparently sparks problems.
> I do not think such a problem will arise in processors classpath
> immediately, but such a situation might happen in future.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/dagger/pull/118/files#r27648909>.
>